pennylane/resources/first_quantization.py

```diff
@@ -15,505 +15,559 @@
 This module contains the functions needed for estimating the number of logical qubits and
 non-Clifford gates for quantum algorithms in first quantization using a plane-wave basis.
 """
-# pylint: disable=too-many-arguments
+# pylint: disable=no-self-use disable=too-many-arguments disable=too-many-instance-attributes
 from scipy import integrate
 from pennylane import numpy as np
+from pennylane.operation import Operation, AnyWires
 
 
-def success_prob(n, br):
-    r"""Return the probability of success for state preparation.
+class FirstQuantization(Operation):
+    r"""Contains the functionality for estimating the number of non-Clifford gates and logical qubits
+    for quantum algorithms in first quantization using a plane-wave basis.
 
-    The expression for computing the probability of success is taken from Eqs. (59, 60) of
-    [`PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_].
+    To estimate the gate and qubit costs for implementing this method, the number of plane waves,
+    the number of electrons and the unit cell volume need to be defined. Once these parameters are
+    defined, the costs can be computed using the functions :func:`~.gate_cost` and
+    :func:`~.qubit_cost` with a target error that has the default value of chemical accuracy
+    (0.0016 Ha) here.
 
-    Args:
-        n (int): number of plane waves
-        br (int): number of bits for ancilla qubit rotation
-
-    Returns:
-        float: probability of success for state preparation
-
-    **Example**
-
-    >>> success_prob(10000, 7)
-    0.9998814293823286
-    """
-    if n <= 0 or not isinstance(n, int):
-        raise ValueError("The number of planewaves must be a positive integer.")
-
-    if br <= 0 or not isinstance(br, int):
-        raise ValueError("br must be a positive integer.")
-
-    c = n / 2 ** np.ceil(np.log2(n))
-    d = 2 * np.pi / 2**br
+    The target algorithm error, :math:`\epsilon`, is distributed among four different sources of
```

Comment:
Let's mention the paper only once. Maybe something like "Following PRX...." and then just explain everything